src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -584,7 +584,43 @@ extern "C" int32_t SystemNative_Link(const char* source, const char* linkTarget)
 extern "C" intptr_t SystemNative_MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     intptr_t result;
+#if HAVE_MKSTEMPS
     while (CheckInterrupted(result = mkstemps(pathTemplate, suffixLength)));
+#elif HAVE_MKSTEMP
+    // mkstemps is not available bionic/Android, but mkstemp is
+    // mkstemp doesn't allow the suffix that msktemps does allow, so we'll need to
+    // remove that before passing pathTemplate to mkstemp
+
+    int32_t pathTemplateLength = static_cast<int32_t>(strlen(pathTemplate));
+
+    if (suffixLength < 0 || suffixLength > pathTemplateLength)
+    {
+        errno = ERANGE;
```

Comment:
Also, I believe the second condition here should be ```suffixLength > pathTemplateLength - 6```, as the pattern needs to contain "XXXXXX" in addition to whatever suffix it contains.